### PR TITLE
Chat bot

### PR DESCRIPTION
**Pull request by Aliiiu**
# Design Chat Bot

This PR works on implementing the chatbot UI

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/2082

### CODE REVIEW

**Review by Simplypheyie96 — COMMENTED** [review]
- This needs to be removed
image
- The UI needs to be updated
image

- another thing that needs update
image

- image

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
- I think there is enough space to make the least font size 12px. on the chatbot.
The components are to make it flexible for all screen sizes. 
if you check the main insert on the wiki page design, you will see the least font size is 12px, please adjust the sizes so they are more visible. they to small right now compared to other widgets (I mentioned this in the video sent). 
nb: let me know if you need me to increase the sizes of the components.
(an image of the insert on wiki page on figma)
image

- The logo has not been updated (also mentioned this in the video
image

- Content really small on the collapsible widget and. there is no margin between text and logo 
image
- presenting data animation is static
image

File: src/components/Wiki/WikiPage/InsightComponents/BrainBot/BotMessages.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  return (
+    <Box
+      display={'flex'}
+      flexDirection={'column'}
+      position={'relative'}
+      gap={'10px'}
+      width={'100%'}
+    >
+      {messages?.map((message) => {
+        console.log(message.answer)
```
Comment (Adebesin-Cell) [inline src/components/Wiki/WikiPage/InsightComponents/BrainBot/BotMessages.tsx:22]: remove logs

File: src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import { RiArrowDownSLine } from 'react-icons/ri'
+import { useAppSelector } from '@/store/hook'
+import { WikiInsightsProps } from '../../WikiInsights'
+import { Wiki } from '@everipedia/iq-utils'
+import ChatBot from './ChatBot'
+import { BrainBotSuggestion } from './BotSuggestions'
+import QuestionMarkIcon from '@/components/Icons/questionMarkIcon'
+import Image from 'next/image'
+
+export const queryMapper = (query: string, wiki: Wiki) => {
```
Comment (Adebesin-Cell) [inline src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx:13]: Kindly move this into a util file and use enums to handle the cases, makes it cleaner.

File: src/components/Wiki/WikiPage/InsightComponents/BrainBot/ChatBot.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}: {
+  wiki: Wiki
+  setOpen: (state: boolean) => void
+}) => {
+  const chatsRef = useRef<HTMLDivElement | null>(null)
+  const { currentHumanMessage, currentChatId, messages } = useAppSelector(
+    (state) => state.message,
+  )
+
+  useEffect(() => {
```
Comment (Adebesin-Cell) [inline src/components/Wiki/WikiPage/InsightComponents/BrainBot/ChatBot.tsx:23]: Extract this logic into a hook and use in places that are duplicated.

Comment (Aliiiu) [reply src/components/Wiki/WikiPage/InsightComponents/BrainBot/ChatBot.tsx:23]: The scrollToBottom logic?

Comment (Adebesin-Cell) [reply src/components/Wiki/WikiPage/InsightComponents/BrainBot/ChatBot.tsx:23]: yes

**Review by Simplypheyie96 — CHANGES_REQUESTED** [review]
- this still looks unbalanced. i think you left a single word in one line, so the logo looks unaligned. (I will change the position of the design so the implementation looks better).
image

- please add hover states for all clickable buttons.
- presenting data animation (loading state) still not working
- for 'ask me about crypto' option on the first page, (I mentioned before), this should show the bot replying "ask me a question about crypto" and not it suggesting questions as answers.
image
 
 - I also realized that the logo is for IQ wiki, so i will be changing all instances of the logo to IQ gpt logo

**Comment by Aliiiu** [comment]
> * this still looks unbalanced. i think you left a single word in one line, so the logo looks unaligned. (I will change the position of the design so the implementation looks better).
> 
> image
> * please add hover states for all clickable buttons.
> * presenting data animation (loading state) still not working
> * for 'ask me about crypto' option on the first page, (I mentioned before), this should show the bot replying "ask me a question about crypto" and not it suggesting questions as answers.
> 
> image
> * All logos used are for IQ wiki instead of IQ gpt. please change them
> * also, like I mentioned before, clicking on the logo should redirect to IQ gpt, so that the function of the IQ GPT banner we had before is maintained.

I don't think it is a good user experience to click on the logo of a widget and it's redirect you to a website. I only added that feature to the footer section of the widget. You can check if the hover state of the footer section is good enough

**Comment by Simplypheyie96** [comment]
> > * this still looks unbalanced. i think you left a single word in one line, so the logo looks unaligned. (I will change the position of the design so the implementation looks better).
> > 
> > image
> > * please add hover states for all clickable buttons.
> > * presenting data animation (loading state) still not working
> > * for 'ask me about crypto' option on the first page, (I mentioned before), this should show the bot replying "ask me a question about crypto" and not it suggesting questions as answers.
> > image
> > * All logos used are for IQ wiki instead of IQ gpt. please change them
> > * also, like I mentioned before, clicking on the logo should redirect to IQ gpt, so that the function of the IQ GPT banner we had before is maintained.
> 
> I don't think it is a good user experience to click on the logo of a widget and it's redirect you to a website. I only added that feature to the footer section of the widget. You can check if the hover state of the footer section is good enough

I did mention previously in the video i sent to you on discord that the logo on the bottom should be the one to redirect to iq gpt and not the whole widget. I was using my cursor in the video to show what redirects to iq gpt. you can watch the video to see
image

File: src/utils/BotUtils/index.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Wiki } from '@everipedia/iq-utils'
+
+enum QueryType {
+  GenerateAdditionalInfo = 'Generate additional info for this page',
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:4]: the enums can be in form of ids perhaps not content text, ``GenerateAdditionalInfo = AdditionalInfo``, just a simple word to describe it.

File: src/utils/BotUtils/index.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  ExplainLikeFive = 'ELI5',
+}
+
+export const queryMapper = (query: string, wiki: Wiki) => {
+  let newQuery = ''
+
+  switch (query) {
+    case QueryType.GenerateAdditionalInfo:
+      newQuery = `${wiki.content} 
+          Generate an additional information for the content above?
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:15]: To make the prompting better, I believe we can use ``Generate additional information for the content delimited by backticks  ${wiki.content} ``

File: src/utils/BotUtils/index.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  switch (query) {
+    case QueryType.GenerateAdditionalInfo:
+      newQuery = `${wiki.content} 
+          Generate an additional information for the content above?        
+        `
+      break
+    case QueryType.ContentPageSummary:
+      newQuery = `${wiki.content} 
+          Summarize the content above?
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:20]: Same with this.

**Review by Simplypheyie96 — APPROVED** [review]

File: src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+          <Box
+            display={'flex'}
+            flexDirection={'column'}
+            gap={'8px'}
+            marginTop={'14px'}
+            alignItems={'center'}
+          >
+            <BrainBotSuggestion
+              question={'Generate additional info for this page'}
```
Comment (Adebesin-Cell) [inline src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx:69]: Not in this PR, but we should add translations for the contents.

File: src/utils/BotUtils/index.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  ContentPageSummary = 'Content/page summary.',
+  ExplainLikeFive = 'ELI5',
+}
+
+export const queryMapper = (query: string, wiki: Wiki) => {
+  let newQuery = ''
+
+  switch (query) {
+    case QueryType.AdditionalInfo:
+      newQuery = `Generate additional information for the content delimited by backticks  ${wiki.content}`
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:14]: Oh you missed to add backticks, it's just to make the focus of the llm better. you can use sth else like hash symbol

File: src/utils/BotUtils/index.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+      break
+    case QueryType.ContentPageSummary:
+      newQuery = `Summarize the content delimited by backticks  ${wiki.content}`
+      break
+    case QueryType.ExplainLikeFive:
+      newQuery = `${wiki.content}
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:20]: duplicated ${wiki.content}, also missing the backticks

File: src/utils/BotUtils/index.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import { Wiki } from '@everipedia/iq-utils'
+
+enum QueryType {
```
Comment (Adebesin-Cell) [inline src/utils/BotUtils/index.ts:3]: Cool, ensure to make the used in other pages so it's easy to edit and update content everywhere.

File: src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx (hunk 7 above)
Comment (Aliiiu) [reply src/components/Wiki/WikiPage/InsightComponents/BrainBot/BrainBot.tsx:69]: Alright

**Review by Adebesin-Cell — APPROVED** [review]
